operators/pkg/controller/apmserver/config/config.go

```diff
@@ -5,189 +5,97 @@
-	ReadTimeout        int       `json:"read_timeout,omitempty"`
-	ShutdownTimeout    string    `json:"shutdown_timeout,omitempty"`
-	SecretToken        string    `json:"secret_token,omitempty"`
-	SSL                TLSConfig `json:"ssl,omitempty"`
-	Rum                RumConfig `json:"rum,omitempty"`
-	ConcurrentRequests int       `json:"concurrent_requests,omitempty"`
-	MaxUnzippedSize    int       `json:"max_unzipped_size,omitempty"`
-}
-
-type RumConfig struct {
-	Enabled   bool `json:"enabled,omitempty"`
-	RateLimit int  `json:"rate_limit,omitempty"`
-}
-
-type TLSConfig struct {
-	Enabled                bool     `json:"enabled"`
-	Certificate            string   `json:"certificate,omitempty"`
-	Key                    string   `json:"key,omitempty"`
-	CertificateAuthorities []string `json:"certificate_authorities,omitempty"`
-}
-
-type LoggingConfig struct {
-	Level          string `json:"level,omitempty"`
-	ToFiles        bool   `json:"to_files,omitempty"`
-	JSON           bool   `json:"json,omitempty"`
-	MetricsEnabled bool   `json:"metrics.enabled,omitempty"`
-}
-
-type LoggingFilesConfig struct {
-	Path      string `json:"path,omitempty"`
-	Name      string `json:"name,omitempty"`
-	Keepfiles int    `json:"keepfiles,omitempty"`
-}
-
-type LoggingMetricsConfig struct {
-	Enabled bool `json:"enabled,omitempty"`
-}
-
-type QueueConfig struct {
-	Mem QueueMemConfig `json:"mem,omitempty"`
-}
-
-type QueueMemConfig struct {
-	Events int         `json:"events,omitempty"`
-	Flush  FlushConfig `json:"flush,omitempty"`
-}
-
-type FlushConfig struct {
-	MinEvents int    `json:"min_events,omitempty"`
-	Timeout   string `json:"timeout,omitempty"`
-}
+	// Build the default configuration
+	defaultCfg, err := settings.ParseConfig(DefaultConfiguration)
+	if err != nil {
+		return nil, err
+	}
 
-type ElasticsearchOutputConfig struct {
-	Hosts            []string          `json:"hosts,omitempty"`
-	SSL              TLSConfig         `json:"ssl,omitempty"`
-	Username         string            `json:"username,omitempty"`
-	Password         string            `json:"password,omitempty"`
-	Headers          map[string]string `json:"headers,omitempty"`
-	Worker           int               `json:"worker,omitempty"`
-	MaxBulkSize      int               `json:"max_bulk_size,omitempty"`
-	CompressionLevel int               `json:"compression_level,omitempty"`
-	Indices          []json.RawMessage `json:"indices,omitempty"`
+	// Merge the configuration with userSettings last so they take precedence.
+	err = cfg.MergeWith(
+		defaultCfg,
+		settings.MustCanonicalConfig(
+			map[string]interface{}{
+				"output.elasticsearch.hosts":                       as.Spec.Output.Elasticsearch.Hosts,
+				"output.elasticsearch.username":                    username,
+				"output.elasticsearch.password":                    password,
+				"output.elasticsearch.ssl.certificate_authorities": []string{"config/elasticsearch-certs/" + certificates.CertFileName},
```

Comment:
`conf/elasticseach-certs` could be a constant we seem to use it in more than one place I believe (the other uses are prefixed with `/user/share/apm-server/` )